src/app/students/hooks/useStudents.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState, useEffect, useCallback, createContext, useContext, ChangeEvent, PropsWithChildren } from 'react'
+import { parseCSV } from '@/lib/parseCSV'
+import { Student } from '@/_temp_types/student'
+
+interface SectionOption {
+  label: string;
+  value: string;
+}
+
+interface StudentsContextType {
+  displayStudents: Student[];
+  currentSections: SectionOption[];
+  handleFileChange: (event: ChangeEvent<HTMLInputElement>) => Promise<void>;
+  handleCancel: () => void;
+  handleSave: () => void;
+}
+
+const StudentsContext = createContext<StudentsContextType | undefined>(undefined)
+
+export const useStudentsProvider = (): StudentsContextType => {
+    const [csvStudentsParse, setStudentsParse] = useState<Student[]>([])
+    const [displayStudents, setDisplayStudents] = useState<Student[]>([])
+    const [currentSections, setSections] = useState<SectionOption[]>([])
+
+    useEffect(() => {
+        const sections = new Set<string>()
+        csvStudentsParse.forEach(student => {
+            student.sections?.forEach(section => {
+                sections.add(section)
+            })
+        })
+        const sectionsOptions: SectionOption[] = Array.from(sections).map(section => ({ label: section, value: section }))
+        setSections(sectionsOptions)
+    }, [csvStudentsParse])
+
+    const handleFileChange = async (event: ChangeEvent<HTMLInputElement>) => {
+        const file = event.target.files?.[0]
+        let parsedCSV: Student[] = []
+        if (file) {
+            parsedCSV = await parseCSV(file) as Student[]
+            setStudentsParse(parsedCSV)
+        }
+    }
+
+    const handleCancel = () => setStudentsParse([])
+
+    const handleSave = () => {
+        // Implement save functionality
+        setDisplayStudents(csvStudentsParse)
+    }
+
+    return {
+        displayStudents,
+        currentSections,
+        handleFileChange,
+        handleCancel,
+        handleSave,
+    }
+}
+
+export const StudentsProvider: React.FC<PropsWithChildren> = ({ children }) => {
+    const students = useStudentsProvider()
```

Comment:
this custom hook isn't needed either, and if it was, `students` is a mega misleading name